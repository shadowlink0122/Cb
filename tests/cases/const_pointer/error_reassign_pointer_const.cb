// エラーテスト: T* const の再代入
include "stdio.cb"

int main() {
    println("=== Test: Cannot reassign const pointer ===");
    int a = 30;
    int b = 40;
    
    int* const ptr = &a;
    
    // エラー: int* const 自体は変更できない
    ptr = &b;  // This should fail
    
    println("ERROR: This should not execute");
    return 0;
}
