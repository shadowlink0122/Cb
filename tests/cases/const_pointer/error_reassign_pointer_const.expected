=== Test: Cannot reassign const pointer ===
Error: Cannot reassign const pointer (T* const)
